### DISCUSSION TITLE
PCAPs not available for requested date?

### DISCUSSION THREAD

**Discussion by lonewaffle** [Security-Onion-Solutions/securityonion#4385] [Unsupported Versions]
New Security Onion install 2.3.51 from ISO, standalone type. Whenever I try to access a PCAP from an alert or the hunting screen, I get the message that the job was unable to complete and underneath it says "No data available for the requested dates". 

In sensoroni.log, I see:
timestamp=2021-06-02T15:26:26.675711227Z level=info message="Skipping steno processor due to date range conflict" availableDataBeginDate=2021-06-02T03:42:13.76519764Z availableDataEndDate=2021-06-02T15:24:26.675708903Z jobBeginDate=2021-06-01T08:31:43.942Z jobEndDate=2021-06-01T09:35:43.942Z jobId=1011
timestamp=2021-06-02T15:26:26.675752527Z level=error message="Failed to process job; job processing aborted" error="No data available for the requested dates" jobId=1011

This may be a retention issue because I'm noticing in the /nsm/pcap directory, all of the items in there only go back less than 24 hours, but I'm also noticing on graphana that there's no data for PCAP retention:
![Screen Shot 2021-06-02 at 9 17 07 AM](https://user-images.githubusercontent.com/5062130/120510567-c7cbfa80-c386-11eb-8836-75b7c03e5c90.png)

If I expand that graph out to 30 days, I can see there was data being graphed for a few days after I first installed SO, but then it abruptly stopped for some reason.

Any help would be appreciated.

**Comment by lonewaffle** [comment]
I'd really appreciate if someone could weigh in on this. Is the error I'm seeing just indicating that the relevant PCAP has already rotated out of storage? Any ideas on why the PCAP Retention graph says "No data"?

**Comment by TOoSmOotH** [comment]
How much space do you have for /nsm?

**Reply by lonewaffle** [reply]
I currently have 6.1TB allocated to NSM and 5.4TB are being used.

![Screen Shot 2021-06-07 at 9 42 49 AM](https://user-images.githubusercontent.com/5062130/121049188-ba42b600-c774-11eb-975c-6a0836ad4f97.png)

When I check the pcap folder, it has stuff in it from the last 10 hours.

**Reply by TOoSmOotH** [reply]
Is this a standalone install? Steno manages its own space and can be modified in the minion pillar. By default, it is set to 10%. Another 5% won't really get you to 24 hours so you either need to reduce the amount of days in Elastic (If it is standalone) or add more disk space.

**Reply by lonewaffle** [reply]
It's standalone. I'm not sure I understand what you mean by "reduce the amount of days in Elastic". Are you saying I should reduce the retention in Elastic in an attempt to make more space for PCAPS? 

Also, do you have any theories on why the graffana graph stopped tracking PCAP retention?

**Reply by lonewaffle** [reply]
Nevermind on the Grafana question. It's randomly decided to start tracking this again.

**Reply by Stephen-Ridgway** [reply]
Hi @lonewaffle, I've had some challenges with pcap retention too. Have you tuned Steno? We found that it was capturing a vast amount of network traffic by default, almost all of which was legitimate traffic. We've used bpfs to filter out 'known good' traffic and ssl/tls encrypted traffic, which brought things under control. As you're running a Standalone node, you'll need to balance the amount of data retained in Elastic (which is set as retention days in the global.sls pillar file, I think) and the amount of data stored as pcaps. I don't think Steno and Elastic interact, so if Elastic has taken a big chunk of disk space, Steno will just fill what's available then start purging the oldest pcaps when the disk is 90% full.

**Reply by lonewaffle** [reply]
Thanks, that's helpful information. Filtering out SSL and also backup related traffic has definitely helped a bit, I still wish there were more granular control for this. I wish we could somehow prioritize keeping PCAPs that are related to an alert for longer, but I'm guessing there's no way to easily do this. 

Definitely not complaining though, Security Onion rocks!

**Comment by c9845** [comment]
@TOoSmOotH I am having a similar issue for viewing of PCAPs for older data.  Per the Grafana console, I have about 3 weeks retention of PCAP files.  However, my /nsm usage is only about 26% out of a total of 3.4T allocated.  How do I increase the retention policy for PCAPs?

**Reply by dougburks** [reply]
@c9845 

Increase the MaxDirectoryFiles setting:
https://docs.securityonion.net/en/2.3/stenographer.html#maxdirectoryfiles

In the future, please start a new discussion instead of replying to somebody else's discussion.

Thanks!
